pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/typedefinition/JavaTypeDefinition.java

```diff
@@ -176,4 +177,34 @@ public JavaTypeDefinition getComponentType() {
 
         return forClass(componentType);
     }
+
+    public boolean isClassOrInterface() {
+        return !clazz.isEnum() && !clazz.isPrimitive() && !clazz.isAnnotation() && !clazz.isArray();
+    }
+
+    public boolean isNullType() {
+        return false;
+    }
+
+    public boolean isPrimitive() {
+        return clazz.isPrimitive();
+    }
+
+    // not an override
+    public boolean equals(JavaTypeDefinition def) {
```

Comment:
Providing proper equality here would be hard as generics have to be examined as well, it may lead to circular reference with simple iteration etc. So overriding the equals method would make putting JavaTypeDefinitions into collections impossible, as they rely on a proper equals method.

I'll rename the method to something more suitable to avoid confusion.